L8_case-study_mandelbrot/Fractal_tier2.sac

```diff
@@ -16,46 +16,34 @@ export all;
  *
  *   @return final complex value
  *****************************************************************************/
-int, complex escapeTimeAndValue(complex z, int depth)
+int, complex escapeTimeAndValue (complex z, int depth)
 {
-  i=0; c=z;
+    i=0; c=z;
 
-  while( (normSq( c) <= 4d) && (i <= depth)) {
-    c=c*c+z; i++;
-  }
+    while( (normSq( c) <= 4d) && (i <= depth)) {
+        c=c*c+z; i++;
+    }
 
-  return(i, c);
+    return(i, c);
 }
 
 /** <!--*******************************************************************-->
  *
- * @fn complex[*] escapeValue(complex[*] z, int depth)
+ * @fn int[m,n], complex[m,n] escapeTimeAndValue (complex[m,n] z, int depth)
  *
  *   @brief maps escapeValue to an entire array of complex numbers
  *
  *****************************************************************************/
 inline
-int[.,.], complex[.,.] escapeTimeAndValue( complex[.,.] plane, int depth)
+int[m,n], complex[m,n] escapeTimeAndValue ( complex[m,n] plane, int depth)
 {
-  times = with {
-            ( . <= x <= .) {
-              t, v = escapeTimeAndValue( plane[x], depth);
-            } : t;
-          } : genarray( shape( plane), 0);
-
-  values = with {
-             ( . <= x <= .) {
-               t, v = escapeTimeAndValue( plane[x], depth);
-             } : v;
-           } : genarray( shape( plane), toc(0));
-
-  return( times, values);
+    return { iv -> escapeTimeAndValue (plane[iv]) };
```

Comment:
Does not compile for 3 reasons: parser cannot handle the implicated 2 return arguments, escapeTimeAndValue also needs depth, and iv will become an int[3] instead of int[2] as complex[m, n] is typedef'd to double[m, n, 2] currently. This works

```
    escape, values = { iv -> escapeTimeAndValue (plane[iv], depth) 
                          | iv < [m, n] };
    return (escape, values);
```